lib/common/avb.c

```diff
@@ -87,47 +93,104 @@ int pci_connect(device_t *igb_dev)
 	return 0;
 }
 
-int gptpinit(int *shm_fd, char **memory_offset_buffer)
+/**
+ * @brief Open the memory mapping used for IPC
+ * @param shm_fd [inout] File descriptor for mapping
+ * @param shm_map [inout] Pointer to mapping
+ * @return 0 for success, negative for failure
+ */
+ 
+int gptpinit(int *shm_fd, char **shm_map)
 {
+	if (NULL == shm_fd || NULL == shm_map) {
+		return -1;
+	}
 	*shm_fd = shm_open(SHM_NAME, O_RDWR, 0);
 	if (*shm_fd == -1) {
 		perror("shm_open()");
-		return false;
+		return -1;
 	}
-	*memory_offset_buffer =
-	    (char *)mmap(NULL, SHM_SIZE, PROT_READ | PROT_WRITE, MAP_SHARED,
-			 *shm_fd, 0);
-	if (*memory_offset_buffer == (char *)-1) {
+	*shm_map = (char *)mmap(NULL, SHM_SIZE, PROT_READ | PROT_WRITE,
+				MAP_SHARED, *shm_fd, 0);
+	if ((char*)-1 == *shm_map) {
 		perror("mmap()");
-		*memory_offset_buffer = NULL;
+		*shm_map = NULL;
 		shm_unlink(SHM_NAME);
-		return false;
+		return -1;
 	}
-	return true;
+	return 0;
 }
 
-void gptpdeinit(int shm_fd, char *memory_offset_buffer)
-{
-	if (memory_offset_buffer != NULL) {
-		munmap(memory_offset_buffer, SHM_SIZE);
+/**
+ * @brief Free the memory mapping used for IPC
+ * @param shm_fd [in] File descriptor for mapping
+ * @param shm_map [in] Pointer to mapping
+ * @return 0 for success, negative for failure
+ *	-1 = close failed
+ *	-2 = munmap failed
+ *	-3 = close and munmap failed
+ */
+
+int gptpdeinit(int *shm_fd, char **shm_map)
```

Comment:
This has a different signature than the one that was in avtp_pipeline; there's an extra pointer indirection for each argument. There's a call in openavb_time_osal.c that needs to be updated.  You should see some warnings related to it when you build it.